libsolidity/formal/SMTChecker.cpp

```diff
@@ -485,21 +485,32 @@ void SMTChecker::endVisit(UnaryOperation const& _op)
 
 		solAssert(isInteger(_op.annotation().type->category()), "");
 		solAssert(_op.subExpression().annotation().lValueRequested, "");
-		if (Identifier const* identifier = dynamic_cast<Identifier const*>(&_op.subExpression()))
+		if (
+			dynamic_cast<Identifier const*>(&_op.subExpression()) ||
```

Comment:
I wonder whether avoiding to duplicate two lines below is worth adding 7 lines for the case distinction inside this combined if... don't you think splitting the two cases might be easier to read :-)?
In any case, it should be fine as it is as well.

Reply:
Can also do that. I basically copied this from the `assignment` visitor which used to do it separate and was recently combined :p

Reply:
Fine with me either way :-).

Reply:
Updated